src/arrayMethodJoin.js

```diff
@@ -4,8 +4,20 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (typeof (this[i]) !== 'undefined' && this[i] !== null) {
```

Comment:
```suggestion
      if (this[i] !== undefined && this[i] !== null) {
```
just compare value, not type